packages/l10n/src/utils.ts

```diff
@@ -1,23 +1,30 @@
+import type { TSupportedLocale } from '../supported-locales';
+// https://babeljs.io/blog/2017/09/11/zero-config-with-babel-macros
+import supportedLocales from /* preval */ '../supported-locales';
```

Comment:
Since the `supported-locales` is a CJS module, it's very tricky to make it so that it can be loaded from both CJS and TS/ESM.

As we do in other places of the codebase, we can use preval to load the module as CJS and inject it the variable here.

Reply:
@emmenko I've seen that we used `/* preval */` couple of times in our code, but wasn’t exactly sure when we use it. 
If I understood correctly, we import `supportedLocales` here and with `prevail` it will be evaluated at build time, right?
Is it used here for performances reason or to make sure imported data remains consistent? Or something else entirely?

Thank you!

Reply:
Thanks for asking.

Most of the time we use preval to perform some operations with the filesystem that would be otherwise impossible within a browser.
The preval macro will evaluate that code and inject it into the bundle.

Here we're leveraging that just for the simplicity of loading a CJS module that would be otherwise tricky to load, as we also need it for nodejs scripts.

I tried different ways of loading the module (as we also need to assign the type declarations to it) but I would get errors like

```
Error: 'default' is not exported by packages/l10n/supported-locales.js, imported by packages/l10n/src/utils.ts
```